metricbeat/module/kubernetes/util/metrics_storage.go

```diff
@@ -0,0 +1,200 @@
+package util
+
+import (
+	"strings"
+	"sync"
+)
+
+type Metrics struct {
+	entries map[Metric]float64
+}
+
+func NewMetrics() *Metrics {
+	ans := &Metrics{
+		entries: make(map[Metric]float64),
+	}
+	return ans
+}
+
+func (m *Metrics) Set(name Metric, value float64) {
+	m.entries[name] = value
+}
+
+func (m *Metrics) Get(name Metric) (float64, bool) {
+	ans, exists := m.entries[name]
+	return ans, exists
+}
+
+func (m *Metrics) GetWithDefault(name Metric, defaultValue float64) (float64, bool) {
+	ans, exists := m.Get(name)
+	if !exists {
+		return defaultValue, false
+	}
+	return ans, exists
+}
+
+func (m *Metrics) Delete(name Metric) {
+	delete(m.entries, name)
+}
+
+func (m *Metrics) Clear() {
+	for k := range m.entries {
+		delete(m.entries, k)
+	}
+}
+
+type MetricSet struct {
+	sync.RWMutex
+	metrics map[string]*Metrics
+}
+
+func NewMetricSet() *MetricSet {
```

Comment:
Whats the meaning of `MetricSet` here? `MetricSet` term is quite special in Beats so this naming here might be confusing.

Reply:
I agree. In the hurry of getting something working fast I couldn't come up with anything better. It's fixed now